src/View/Helper/BreadcrumbsHelper.php

```diff
@@ -0,0 +1,277 @@
+     * @return $this
+     */
+    public function insertAt($index, $title, $link = null, array $options = [])
+    {
+        array_splice($this->crumbs, $index, 0, [['title' => $title, 'link' => $link, 'options' => $options]]);
+
+        return $this;
+    }
+
+    /**
+     * Insert a crumb before the first matching crumb with the specified title.
+     *
+     * Finds the index of the first middleware that matches the provided class,
+     * and inserts the supplied callable before it.
+     *
+     * @param string $matchingTitle The title of the crumb you want to insert this one before
+     * @param string $title Title of the crumb
+     * @param string|array|null $link Link of the crumb. Either a string, an array of route params to pass to
+     * Url::build() or null / empty if the crumb does not have a link
+     * @param array $options Array of options
+     * @return $this
+     */
+    public function insertBefore($matchingTitle, $title, $link = null, array $options = [])
+    {
+        $key = $this->findCrumb($matchingTitle);
+
+        if ($key !== null) {
+            return $this->insertAt($key, $title, $link, $options);
+        }
+        throw new LogicException(sprintf("No crumb matching '%s' could be found.", $matchingTitle));
+    }
+
+    /**
+     * Insert a crumb after the first matching crumb with the specified title.
+     *
+     * Finds the index of the first middleware that matches the provided class,
+     * and inserts the supplied callable before it.
+     *
+     * @param string $matchingTitle The title of the crumb you want to insert this one after
+     * @param string $title Title of the crumb
+     * @param string|array|null $link Link of the crumb. Either a string, an array of route params to pass to
+     * Url::build() or null / empty if the crumb does not have a link
+     * @param array $options Array of options
+     * @return $this
+     */
+    public function insertAfter($matchingTitle, $title, $link = null, array $options = [])
+    {
+        $key = $this->findCrumb($matchingTitle);
+
+        if ($key !== null) {
+            return $this->insertAt($key + 1, $title, $link, $options);
+        }
+        throw new LogicException(sprintf("No crumb matching '%s' could be found.", $matchingTitle));
+    }
+
+    /**
+     * Returns the crumbs list
+     *
+     * @return array
+     */
+    public function getCrumbs()
+    {
+        return $this->crumbs;
+    }
+
+    /**
+     * Renders the breadcrumbs trail
+     *
+     * @param array $attributes Array of attributes applied to the wrapper element
+     * @param array $separator Array of attributes for the `separator` template
+     * @return string The breadcrumbs trail
+     */
+    public function render(array $attributes = [], array $separator = [])
+    {
+        $crumbs = $this->crumbs;
+        $crumbsCount = count($crumbs);
+        $templater = $this->templater();
+
+        $separatorParams = [];
+        if (!empty($separator)) {
```

Comment:
I am not sure its a good idea for non-booleans either. But then `empty()` has its side effects, too.